p2p/peer_stats_test.go

```diff
@@ -110,3 +113,58 @@ func Test_StatDecreaseScore(t *testing.T) {
 	pStats.decreaseScore()
 	require.Equal(t, pStats.score(), float32(80.0))
 }
+
+// BenchmarkPeerQueue/push_for_10-14         	 2373187	       489.3 ns/op	     448 B/op	       8 allocs/op
+// BenchmarkPeerQueue/pop_for_10-14          	 1561532	       761.8 ns/op	     800 B/op	      10 allocs/op
+// BenchmarkPeerQueue/push_for_100-14        	  253057	      4590 ns/op	    2368 B/op	      11 allocs/op
+// BenchmarkPeerQueue/pop_for_100-14         	  151729	      7503 ns/op	    8000 B/op	     100 allocs/op
+// BenchmarkPeerQueue/push_for_1000-14       	   25915	     46220 ns/op	   17728 B/op	      14 allocs/op
+// BenchmarkPeerQueue/pop_for_1000-14        	   15207	     75764 ns/op	   80000 B/op	    1000 allocs/op
+// BenchmarkPeerQueue/push_for_1000000-14    	      15	  69555594 ns/op	44948820 B/op	      41 allocs/op
+// BenchmarkPeerQueue/pop_for_1000000-14     	      15	  75170956 ns/op	80000032 B/op	 1000000 allocs/op
+func BenchmarkPeerQueue(b *testing.B) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	peers := [][]*peerStat{
+		generatePeerStats(b, 10),
+		generatePeerStats(b, 100),
+		generatePeerStats(b, 1000),
+		generatePeerStats(b, 1000000),
+	}
+
+	for _, peerStats := range peers {
+		var queue *peerQueue
```

Comment:
I think keeping `queue` inside every subtest will be better to misuse it in the future (assume moving 1 subtest before another or so).